### ISSUE TITLE
Relocation after 'kidnap'? 

### ISSUE THREAD

**Issue by antithing** [ethz-asl/okvis#73]
Does okvis handle relocation if tracking is lost? For example, if the sensor is blocked and moved, when it is unblocked, will it localize correctly in the map? (if it is still in the pre mapped region, obviously).

Thanks!

**Comment by NikolausDemmel** [comment]
No, it's a keyframe odometry, not a SLAM system. IMU might allow it to track for a short while without usable camera input though.

**Comment by antithing** [comment]
Thank you! Understood. I might try to add this function in. Would you have any pointers on where to start?

**Comment by NikolausDemmel** [comment]
I suggest to read and understand the related publications.

IMHO it is not obvious how to extend the marginalization sliding window to include old keyframes after loop closure or relocalization. What might work is having a two-layer system. The current sliding window as a front-end odometry and then after that a mapping module with relocalization, loop closure detection, pose graph optimization and possibly bundle adjustment (similar to orbslam, possibly with stereo and IMU error terms in the BA). However, the mapping module would never directly influence the frontend odometry, but instead only update it's "reference frame". Overall it seems like a quite involved project to me.
